unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/WebRequestController.cs

```diff
@@ -51,12 +53,164 @@ public static WebRequestController Create()
+            Action<UnityWebRequest> onSuccess = null,
+            Action<UnityWebRequest> onfail = null,
+            int requestAttemps = 3,
+            int timeout = 0,
+            CancellationToken cancellationToken = default,
+            Dictionary<string, string> headers = null)
+        {
+            postWebRequestFactory.SetBody(postData);
+            return await SendWebRequest(postWebRequestFactory, url, downloadHandler, onSuccess, onfail, requestAttemps,
+                timeout, cancellationToken, headers);
+        }
+
+        public async UniTask<UnityWebRequest> GetAssetBundleAsync(
+            string url,
+            Action<UnityWebRequest> onSuccess = null,
+            Action<UnityWebRequest> onfail = null,
+            int requestAttemps = 3,
+            int timeout = 0,
+            CancellationToken cancellationToken = default)
+        {
+            return await SendWebRequest(assetBundleFactory, url, null, onSuccess, onfail, requestAttemps,
+                timeout, cancellationToken);
+        }
+
+        public async UniTask<UnityWebRequest> GetAssetBundleAsync(
+            string url,
+            Hash128 hash,
+            Action<UnityWebRequest> onSuccess = null,
+            Action<UnityWebRequest> onfail = null,
+            int requestAttemps = 3,
+            int timeout = 0,
+            CancellationToken cancellationToken = default)
+        {
+            assetBundleFactory.SetHash(hash);
+            return await SendWebRequest(assetBundleFactory, url, null, onSuccess, onfail, requestAttemps,
+                timeout, cancellationToken);
+        }
+
+        public async UniTask<UnityWebRequest> GetTextureAsync(
+            string url,
+            Action<UnityWebRequest> onSuccess = null,
+            Action<UnityWebRequest> onfail = null,
+            int requestAttemps = 3,
+            int timeout = 0,
+            bool isReadable = true,
+            CancellationToken cancellationToken = default,
+            Dictionary<string, string> headers = null)
+        {
+            textureFactory.isReadable = isReadable;
+            return await SendWebRequest(textureFactory, url, null, onSuccess, onfail, requestAttemps,
+                timeout, cancellationToken, headers);
+        }
+
+        public async UniTask<UnityWebRequest> GetAudioClipAsync(
+            string url,
+            AudioType audioType,
+            Action<UnityWebRequest> onSuccess = null,
+            Action<UnityWebRequest> onfail = null,
+            int requestAttemps = 3,
+            int timeout = 0,
+            CancellationToken cancellationToken = default)
+        {
+            audioClipWebRequestFactory.SetAudioType(audioType);
+            return await SendWebRequest(audioClipWebRequestFactory, url, null, onSuccess, onfail, requestAttemps,
+                timeout, cancellationToken);
+        }
+        
+
+        private async UniTask<UnityWebRequest> SendWebRequest<T>(
```

Comment:
I kinda agree that we don't need `disposeOnCompleted` here as it can be handled by consumers simply by wrapping in `using`. But was it intentional?

Reply:
It was 100% intentional